### PR TITLE
Cryopod antirobbery system

### PR DESCRIPTION
**Pull request by soslow4u**
Discuss

### CODE REVIEW

**Comment by GrayRachnid** [comment]
Speedmerge

**Comment by Kurgis** [comment]
Cryopod only removes the occupant from the round into cryogenic storage once the occupant either ghosts or logs off. Far as I'm aware, cryopods can't be opened or interacted with, so someone can jump into a cryopod to be effectively invulnerable. Correct me if I'm wrong, but otherwise, there's no possible way to remove someone from a cryopod except by ejecting the occupant.

**Comment by Kurgis** [comment]
The cryopod lock should check whether the occupant is logged on, and still in their body. It should only lock if the person is ghosted, offline, or otherwise outside of their character's body.

**Comment by soslow4u** [comment]
Resolved

File: code/game/machinery/cryopod.dm [hunk 1]
```diff
@@ -455,6 +455,9 @@
 	//Eject any items that aren't meant to be in the pod.
 	var/list/items = src.contents
 	if(occupant)
+		if(usr != occupant && !occupant.client && occupant.stat<2)
```
Comment (mattroks101) [inline code/game/machinery/cryopod.dm]: What is the stat greater than 2?

Comment (mattroks101) [reply code/game/machinery/cryopod.dm]: Like, stat 2 is dead, do we have a special one for stuff greater than 2?

Comment (soslow4u) [reply code/game/machinery/cryopod.dm]: Well, look at this one
https://github.com/discordia-space/CEV-Eris/blob/9ea20b5c4dc30d9cc0f4caaa6abbee2067b5d754/code/game/machinery/cryopod.dm#L254-L269

Comment (mattroks101) [reply code/game/machinery/cryopod.dm]: Oh I'm really fucking dumb. I misread it. God I hate this stat shit sometimes.

Comment (LethalGhost) [inline code/game/machinery/cryopod.dm]: Related to previous conversation
>Occupant is living and has no client.

actually means `occupant.stat != DEAD` ~~(or may be `! (occupant.stat & DEAD)`)~~
Anyway 2 is magick number.
```suggestion
		if(usr != occupant && !occupant.client && occupant.stat != DEAD)
```

Comment (soslow4u) [reply code/game/machinery/cryopod.dm]: Well, I tried to reuse code, as calling `incapacitated()` would be abused with just using `sleep` verb
https://github.com/discordia-space/CEV-Eris/blob/907d44d858d5fe0644724a7b8da7936f73e9eac1/code/__defines/mobs.dm#L154-L166
As there is no value for DEAD case. Honestly, have no idea how would it check if mob is dead and took the easiest option was available

Comment (LethalGhost) [reply code/game/machinery/cryopod.dm]: Check "Suggested change".

Comment (TorinoFermic) [reply code/game/machinery/cryopod.dm]: This is what I used on the excelsior chat to share the contents to ghosts with AntagHUD enabled.

**Pull request merged**